components/server/src/projects/projects-service.ts

```diff
@@ -92,7 +93,32 @@ export class ProjectsService {
             ...(!!userId ? { userId } : { teamId }),
             appInstallationId
         });
-        return this.projectDB.storeProject(project);
+        await this.projectDB.storeProject(project);
+        await this.onDidCreateProject(project);
+        return project;
+    }
+
+    protected async onDidCreateProject(project: Project) {
+        let { userId, teamId, cloneUrl } = project;
+        const parsedUrl = parseRepoUrl(project.cloneUrl);
+        if ("gitlab.com" === parsedUrl?.host) {
```

Comment:
It should work with self-managed gitlab's as well.

Reply:
**fyi**: There's a follow up issue for self hosted support in https://github.com/gitpod-io/gitpod/issues/5115. Cc @svenefftinge